tests/kafkatest/services/performance/producer_performance.py

```diff
@@ -62,6 +63,7 @@ def parse_stats(line):
             }
         last = None
         for line in node.account.ssh_capture(cmd):
+            self.maybe_start_jmx_tool(idx, node)
```

Comment:
@lindong28 Now it's a lot clearer what you meant in your original email. You're basically saying that since `RemoteAccount.ssh()` blocks, `RemoteAccount.ssh_output()` blocks until all output is collected and `RemoteAccount.ssh_capture()` drives the whole process despite acting as a generator, there's no way to start a second process between when you start the first and when you start reading its output?

I think it makes a lot of sense to provide support for that. More generally, I think the current set of `RemoteAccount.ssh*` methods could use some cleanup and potentially renaming to simplify things. Originally I thought the blocking `RemoteAccount.ssh()` would be the most widely used, but it turns out we've found more and more uses for capturing the output.

@granders, a patch just to add Ducktape support for the approach @lindong28 is looking for should be quick and easy. I think it can look a lot like `ssh_capture`, but it'll have to return an iterable object after invoking the ssh command rather than treating the entire method call as a generator using yield. It might even make sense to just return the subprocess object and allow the user to do whatever they want with it, so the example would look something like

```
cmd = "same_as_before"
proc = node.account.ssh_async(cmd)
self.start_jmx_tool(node)
for line in iter(proc.stdout.readline, ''):
   # same as the previous for loop
```

I think there are cleaner ways to expose this, but given how many `ssh*` variants we're ending up with, we may want to hold off on introducing much more until we sort out how to capture all of them with a minimum of methods.